src/ServiceControl.MultiInstance.AcceptanceTests/Connection/PlatformConnectionTests.cs

```diff
@@ -0,0 +1,140 @@
+    using NServiceBus.AcceptanceTesting;
+    using NUnit.Framework;
+    using Particular.Approvals;
+    using TestSupport;
+    using TestSupport.EndpointTemplates;
+    using ServiceControl.AcceptanceTesting;
+    using ServiceControlInstaller.Engine.Instances;
+
+    [RunOnAllTransports]
+    class PlatformConnectionTests : AcceptanceTest
+    {
+        [Test]
+        public async Task ExposesConnectionDetails()
+        {
+            var config = await Define<MyContext>()
+                .WithEndpoint<MyEndpoint>()
+                .Done(async x =>
+                {
+                    var result = await this.GetRaw("/api/connection", ServiceControlInstanceName);
+                    x.Connection = await result.Content.ReadAsStringAsync();
+                    return true;
+                })
+                .Run();
+
+            Assert.IsNotNull(config.Connection);
+
+            var formatted =
+                JsonConvert.SerializeObject(
+                    JsonConvert.DeserializeObject(config.Connection),
+                    Formatting.Indented
+                );
+
+            Approver.Verify(
+                formatted,
+                scenario: ScenarioName,
+                scrubber: Scrub
+            );
+        }
+
+        string Scrub(string input)
+        {
+            // MSMQ
+            var result = input.Replace(
+                Environment.MachineName,
+                "MACHINE_NAME"
+            );
+
+            if (!string.IsNullOrWhiteSpace(TransportIntegration.ConnectionString))
+            {
+                try
+                {
+                    var builder = new DbConnectionStringBuilder { ConnectionString = TransportIntegration.ConnectionString };
+
+                    // SQS
+                    if (builder.TryGetValue("QueueNamePrefix", out var queueNamePrefix))
+                    {
+                        var queueNamePrefixAsString = (string)queueNamePrefix;
+                        if (!string.IsNullOrEmpty(queueNamePrefixAsString))
+                        {
+                            result = result.Replace(
+                                queueNamePrefixAsString,
+                                "queue-prefix-"
+                            );
+                        }
+                    }
+
+                    // SQL
+                    if (builder.TryGetValue("Database", out var database))
+                    {
+                        var databaseAsString = (string)database;
+                        if (!string.IsNullOrEmpty(databaseAsString))
+                        {
+                            result = result.Replace(
+                                $"[{databaseAsString}]",
+                                "[DATABASE]"
+                            );
+                        }
+                    }
+                }
+                catch
+                {
+                    // NOTE: Learning Transport has a connection string in an invalid format
```

Comment:
Should we detect LearningTransport explicitly and exit `Scrub` early in such a case?

Reply:
Moved the scrubber to the transport integration classes so each one can provide only the scrubbing needed for it's own idiosyncracies